utf8proc.c

```diff
@@ -165,12 +168,9 @@ UTF8PROC_DLLEXPORT utf8proc_ssize_t utf8proc_encode_char(utf8proc_int32_t uc, ut
     dst[0] = 0xC0 + (uc >> 6);
     dst[1] = 0x80 + (uc & 0x3F);
     return 2;
-  } else if (uc == 0xFFFF) {
-    dst[0] = 0xFF;
-    return 1;
-  } else if (uc == 0xFFFE) {
-    dst[0] = 0xFE;
-    return 1;
+  } else if (((utf8proc_uint32_t)uc)-0xd800 < 0x800) {
+    // Surrogates not valid code point, can't be encoded!
+    return 0;
```

Comment:
OK, my take on this, is that for *validity* checking (unless we have some optional flags to make it relaxed... default strict), it should always be strict.  I think only conversion functions should potentially be relaxed, and then only for the two semi-standard variants, Modified UTF-8 (i.e. like Java), and CESU-8, like
Oracle, MySQL, and others (and that the conversion functions in Base only *produce* 100% correct Unicode... if somebody wants to produce something like Modified UTF-8 or CESU-8, they can easily write that in a package...)